### ISSUE TITLE
Implement automatic renaming of duplicate instance names

### ISSUE THREAD

**Issue by Feuermurmel** [jstasiak/python-zeroconf#29]
Bonjour automatically renames DNS-SD instances with conflicting names. E.g if I use two instances of the `dns-sd` command line tool to register two instances with the same name, the first one will show this:

```
michi ~$ dns-sd -R test _foo '' 10001
Registering Service test._foo._tcp port 10001
DATE: ---Mon 28 Sep 2015---
 1:06:49.909  ...STARTING...
 1:06:50.827  Got a reply for service test._foo._tcp.local.: Name now registered and active
```

While The second one will show this:

```
michi ~$ dns-sd -R test _foo '' 10002
Registering Service test._foo._tcp port 10002
DATE: ---Mon 28 Sep 2015---
 1:06:55.814  ...STARTING...
 1:06:56.990  Got a reply for service test (2)._foo._tcp.local.: Name now registered and active
```

I.e. the instance has been renamed to "test (2)".

I think it would be great if that feature was also added to this library.

**Comment by stephenrauch** [comment]
There is already code present to do this sort of thing, but it broken, so this is actually a bug, not a feature request.

**Comment by stephenrauch** [comment]
Fixed via #75
